lib/katello/tasks/fix_hostgroup_facets.rake

```diff
@@ -0,0 +1,39 @@
+namespace :katello do
+  desc "This task collates hostgroup content facts that were missed during the upgrade from audit.\
+        It then updates the hostgroup content_facet accordingly."
+  task :fix_hostgroup_facets => :environment do
+    def interested_hostgroups
+      Hostgroup.unscoped.where(id: Katello::Hostgroup::ContentFacet.
+                          where(content_source_id: nil,
+                                kickstart_repository_id: nil,
+                                content_view_id: nil,
+                                lifecycle_environment_id: nil).select(:hostgroup_id))
+    end
+
+    def pick_facet_values(hg)
```

Comment:
This call looks at the audit logs for a host group. Pries out information related to lce, ks, cv and content_source_id from the audit logs.
The audit logs typically only contain updates. So if the user changed  just the content_view_id, then that is the only thing marked as audited_changes. Hence we need to go through all the audit logs until we have information on lce, ks, cv and cs. 
Now if there was only one audit log and that was during the creation of hostgroup the audited changes look like this
```ruby
{
 content_view_id: 10,
 kickstart_repository_id: 1000
 ......
}
```
However if you updated the hostgroup and set the kickstart_repository_id, or content_view_id then audited changes look like

```ruby
{
 content_view_id: [10, 11],
 kickstart_repository_id: [1000, 1200]
 ......
}
```
So the code says "if the attribute value is an array pick the last value else just keep the value as it is " 

Further along it is to be noted that `hostgroup.audits` returns the audits ordered by the version number in ascending order, so the latest audit will be `hostgroup.audits.last`

We want to iterate though each audit from latest audit to start, and as soon as we find a  content_view_id key or kickstart_repository_id key or lifecycle environment_id key  or content_source_id key we want it to be set once.

So if I had an audit history like
``` ruby
{
 content_view_id: 10,
 kickstart_repository_id: 1000,
 version:1
 ......
},
{
 content_view_id: [10, 11],
 kickstart_repository_id: [1000, 1200],
  version: 2
 ......
}
```

The code would start at version 2, notice that cv_id and ks_repo were set there and keep them as the final. So when it goes to version 1 since cv_id and ks_repo are already set, it will ignore. It will finally return ```{content_view_id: 11, kickstart_repository_id: 1200} ```